documents/README.md

```diff
@@ -0,0 +1,176 @@
+# Documents
+
+> Create new documents and make them collaborative.
+
+![Documents example](./preview.gif)
+
+> ⚠ **This example only works on JupyterLab v3.1 or higher**
+
+> Before starting this guide, we strongly recommend to look at the documentation, precisely the section of [Documents](https://jupyterlab.readthedocs.io/en/stable/extension/documents.html#documents)
+
+- [Documents](#documents)
+  - [Introduction to documents](#introduction-to-documents)
+  - [Factories](#factories)
+  - [Registering new Documents](#registering-new-documents)
+  - [Document Widget](#document-widget)
+  - [Document Model](#document-model)
+  - [Shared Model](#shared-model)
+
+## Introduction to documents
+
+In JupyterLab, we refer to a document to those widgets backed by a file stored on disk. These files are represented in the frontend by a `Context` which is the bridge between the file and its model, `DocumentModel` representing the data in the file and `DocumentWidget`, which is the view of the model. To make the documents API extensible to enable other developers to write new extensions to support different file types, JupyterLab introduces the `DocumentRegistry` to register new `FileType`s, `DocumentModel`s and `DocumentWidget`s. This way, when opening a new file, the `DocumentManager` will look into the file metadata and create an instance of `Context` with the right `DocumentModel` for this file. To register new documents, you can create factories, either a `ModelFactory` for the model or a `WidgetFactory` for the view.
+
+## Factories
+
+Factories are objects meant to create instances of the suitable Widget/Model given a file. For example, when the `DocumentManager` detects that the file is a notebook, it uses the notebook widget factory to create a new instance of `NotebookPanel`. On the other hand, if you want to make a new `DocumentModel` or `DocumentWidget` for a specific file type, you have to create a factory and register it to the `DocumentRegister`. When registering a factory, you tell the `DocumentManager` that you added a new Model or Widget for a specific file. Then, the `DocumentManager` will use those factories to create instances of the new `DocumentModel` or `DocumentWidget`.
```

Comment:
Format should be consistent, if you use write you refer a code class name, the word should match the code and be formatted with <code>`</code>. Otherwise use lower case and if you want to emphase a word use _italic_ format.

> This remark is mainly for _widget_ and _model_ word.